### PR TITLE
refactor(ui): use smart input field to replace all password sign-in form

### PR DESCRIPTION
**Pull request by simeng-li**
### use the smart input field to replace all password sign-in forms.

This PR includes the following updates:

- fix naming issue in phrases: `username_valid_charset` -> `username_invalid_charset`
- add new phrases: `general_required` and `general_invalid`. Shows the error message for the smart input field with one or more enabled identifier types.
   image. image
- Add new `PasswordSignInForm` containers that integrate `SmartInputField`, `PasswordInputField`, `Terms`, and submit button.
- remove the original `UserSignIn`, `EmailPassword`, and `PhonePassword` forms.

## Testing

test locally
UT added

### CODE REVIEW

**Review by wangsijie — APPROVED** [review]

File: packages/ui/src/pages/SecondarySignIn/index.tsx [hunk 1]
```diff
@@ -31,9 +32,7 @@ const SecondarySignIn = () => {
         <PhoneSignIn autoFocus signInMethod={signInMethod} />
       ) : signInMethod.identifier === SignInIdentifier.Email ? (
         <EmailSignIn autoFocus signInMethod={signInMethod} />
-      ) : (
-        <UsernameSignIn autoFocus />
-      )}
+      ) : null}
```
Comment (charIeszhao) [inline packages/ui/src/pages/SecondarySignIn/index.tsx:35]:
```
{signInMethod.identifier === SignInIdentifier.Phone && <PhoneSignIn autoFocus signInMethod={signInMethod} />}
{signInMethod.identifier === SignInIdentifier.Email && <EmailSignIn autoFocus signInMethod={signInMethod} />}
```

File: packages/ui/src/utils/form.ts [hunk 2]
```diff
@@ -11,3 +26,35 @@ export const passwordErrorWatcher = (error?: FieldError): ErrorType | undefined
...
+  switch (error?.type) {
+    case 'required':
+      return {
+        code: 'general_required',
+        data: { types: enabledFields.map((field) => t(identifierInputPlaceholderMap[field])) },
+      };
+    case 'validate':
+      return {
+        code: 'general_invalid',
+        data: { types: enabledFields.map((field) => t(identifierInputPlaceholderMap[field])) },
```
Comment (charIeszhao) [inline packages/ui/src/utils/form.ts:43]: nit: You can declare `const data` first and use it in both cases

File: packages/ui/src/utils/field-validations.ts [hunk 3]
```diff
@@ -22,7 +27,29 @@ export const usernameValidation = (username: string): ErrorType | undefined => {
...
+  }
+};
+
+export const emailValidation = (email: string): ErrorType | undefined => {
+  if (!emailRegex.test(email)) {
+    return 'invalid_email';
+  }
+};
+
+export const phoneValidation = (value: string): ErrorType | undefined => {
```
Comment (charIeszhao) [inline packages/ui/src/utils/field-validations.ts:40]: Again, function name should be verb-object phrase. Thus, `validateEmail` and `validatePhone`

File: packages/ui/src/utils/field-validations.ts [hunk 4]
```diff
@@ -1,7 +1,12 @@
...
+
 import type { ErrorType } from '@/components/ErrorMessage';
+import { parseE164Number } from '@/utils/country-code';
+
+// TODO: clean up this file and merge to form utils
 
-const usernameRegex = /^[A-Z_a-z-][\w-]*$/;
-const emailRegex = /^\S+@\S+\.\S+$/;
+export const usernameRegex = /^[A-Z_a-z-][\w-]*$/;
+export const emailRegex = /^\S+@\S+\.\S+$/;
```
Comment (charIeszhao) [inline packages/ui/src/utils/field-validations.ts:9]: Can we import these regex from schemas?

File: packages/ui/src/pages/SecondarySignIn/index.tsx (hunk 1 above)
Comment (simeng-li) [reply packages/ui/src/pages/SecondarySignIn/index.tsx:35]: This page will be deleted

File: packages/ui/src/utils/form.ts [hunk 5]
```diff
@@ -11,3 +26,35 @@ export const passwordErrorWatcher = (error?: FieldError): ErrorType | undefined
...
+      };
+    default:
+  }
+};
+
+export const identifierFieldValidation = (type: IdentifierInputType, value: string) => {
```
Comment (charIeszhao) [inline packages/ui/src/utils/form.ts:49]: validateIdentifier

File: packages/ui/src/utils/form.ts [hunk 6]
```diff
@@ -11,3 +26,37 @@ export const passwordErrorWatcher = (error?: FieldError): ErrorType | undefined
...
+        data,
+      };
+    case 'validate':
+      return {
+        code: 'general_invalid',
+        data,
+      };
+    default:
+  }
+};
```
Comment (charIeszhao) [inline packages/ui/src/utils/form.ts:49]: Would be better to use a hook for better performance. e.g. useMemo

Comment (simeng-li) [reply packages/ui/src/utils/form.ts:49]: will be removed in the following PR.

**Pull request merged**